test/e2e/webhook_e2e_test.go

```diff
@@ -0,0 +1,529 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	v1 "github.com/operator-framework/api/pkg/operators/v1"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/install"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
+)
+
+// Global Variables
+const (
+	webhookName = "webhook.test.com"
+)
+
+var _ = Describe("CSVs with a Webhook", func() {
+	var c operatorclient.ClientInterface
+	var crc versioned.Interface
+	var namespace *corev1.Namespace
+	var nsCleanupFunc cleanupFunc
+	BeforeEach(func() {
+		c = newKubeClient(GinkgoT())
+		crc = newCRClient(GinkgoT())
+		namespace, nsCleanupFunc = newNamespace(GinkgoT(), c, genName("webhook-test-"))
+	})
+	AfterEach(func() {
+		nsCleanupFunc()
+	})
+	When("Installed in an OperatorGroup that defines a selector", func() {
+		var nsLabels = map[string]string{
+			"foo": "bar",
+		}
+		var cleanupCSV cleanupFunc
+		var ogSelector = &metav1.LabelSelector{
+			MatchLabels: nsLabels,
+		}
+		var err error
+		BeforeEach(func() {
+			// Add a label to the namespace
+			err = wait.PollImmediate(time.Second*5, time.Minute*5, func() (bool, error) {
```

Comment:
Gomega offers [Eventually](https://onsi.github.io/gomega/#eventually) which could be used instead of `wait.Poll`
See [example](https://github.com/operator-framework/operator-lifecycle-manager/blob/051772d2e0816359e7c45122c52917ab4cf04e57/test/e2e/scoped_client_test.go#L138)
Note when using `Eventually`:
1. It takes polling interval and timeout as an optional argument. By default, all our ginkgo tests use this [default setting](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/test/e2e/e2e_test.go#L45)
2. Also, be sure to read how `Eventually` behaves for [multiple return types](https://github.com/onsi/gomega/blob/master/gomega_dsl.go#L193)